src/openassetio-core/include/openassetio/hostApi/EntityReferencePager.hpp

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2023 The Foundry Visionmongers Ltd
+
+#pragma once
+
+#include <memory>
+
+#include <openassetio/EntityReference.hpp>
+#include <openassetio/typedefs.hpp>
+
+OPENASSETIO_FWD_DECLARE(managerApi, HostSession)
+OPENASSETIO_FWD_DECLARE(managerApi, EntityReferencePagerInterface)
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace hostApi {
+
+OPENASSETIO_DECLARE_PTR(EntityReferencePager)
+
+/**
+ * The EntityReferencePager is the Host facing representation of a
+ * @fqref{managerApi.EntityReferencePagerInterface} implementation.
+ * The Manager class shouldn't be directly constructed by the host.
+ * The EntityReferencePager allows for the retrieval and traversal of large datasets
+ * in a paginated manner.
+ *
+ * Due to the variance of backends, construction, `hasNext`, `get` and
+ * `next` may all reasonably need to perform non trivial, networked
+ * operations, and thus performance characteristics should not be
+ * assumed.singleEntityReferencePager<Elem>
+ *
+ * Destruction of this object is a signal to the manager that the
+ * connection query is finished. For this reason you should avoid
+ * keeping hold of this object for longer than necessary.
+ */
+class OPENASSETIO_CORE_EXPORT EntityReferencePager {
```

Comment:
[consider] `final`?

Reply:
For me, `final` conveys intent in the public interface, and could perhaps avoid a manager implementation attempting to derive the wrong class, the `make_shared` trick feels to be a private implementation detail that makes us feel better.

Reply:
Don't personally mind using the raw shared ptr constructor, even without final it seems like the constructable thing is more prone to confusion.

Added final, removed constructible trick : https://github.com/OpenAssetIO/OpenAssetIO/pull/986/commits/1b8d6f0affb98a575470bfc0082c3e9cc947b001